node/core/pvf-checker/src/lib.rs

```diff
@@ -536,9 +537,30 @@ async fn initiate_precheck(
 ) {
 	gum::debug!(target: LOG_TARGET, ?validation_code_hash, ?relay_parent, "initiating a precheck",);
 
+	let (executor_params_tx, executor_params_rx) = oneshot::channel();
+	sender
+		.send_message(RuntimeApiMessage::Request(
+			relay_parent,
+			RuntimeApiRequest::SessionExecutorParams(executor_params_tx),
+		))
+		.await;
+
+	// FIXME: Error checking?
+	let executor_params = match executor_params_rx.await {
+		Err(_) => return,
+		Ok(Err(_)) => return,
+		Ok(Ok(Some(eep))) => eep,
+		Ok(Ok(None)) => return,
+	};
```

Comment:
I'm rusty on PVF-prechecking. What are the implications of an early return here?

Reply:
I'm not sure myself and I'm still trying to draw @pepyakin's attention to that for him to explain how that part works :)

Reply:
I came through it myself and I hope I've managed to make it right.